Source/SwiftLintFramework/Protocols/CallPairRule.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  CallPairRule.swift
+//  SwiftLint
+//
+//  Created by Tom Quist on 11/07/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import SourceKittenFramework
+
+public protocol CallPairRule: Rule {}
+
+extension CallPairRule {
+
+    public func validate(file: File,
+                         pattern: String,
+                         patternSyntaxKind: [SyntaxKind],
+                         callNameSuffix: String,
+                         severity: ViolationSeverity) -> [StyleViolation] {
+        let firstRanges = file.match(pattern: pattern, with: patternSyntaxKind)
+        let contents = file.contents.bridge()
+        let structure = file.structure
+
+        let violatingLocations: [Int] = firstRanges.flatMap { range in
+            guard let bodyByteRange = contents.NSRangeToByteRange(start: range.location,
+                                                                  length: range.length),
+                case let firstLocation = range.location + range.length - 1,
+                let firstByteRange = contents.NSRangeToByteRange(start: firstLocation,
+                                                                 length: 1) else {
+                                                                    return nil
```

Comment:
The general style used elsewhere in this project is to indent the body of `if`/`guard`/block statements by one indentation level more than the parent level. I know Xcode makes this difficult, but I'd like to stay consistent with the current style for now.

Same with the call to `methodCall` below.